test/test_basic.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import base64
+import unittest
+import subprocess
+import os
+import time
+import urllib.error
+import urllib.request
+import ssl
+import json
+
+
+dirname = os.path.dirname(__file__)
+projroot = os.path.join(dirname, "..")
+
+
+class IntegrationTest(unittest.TestCase):
+
+    def setUp(self):
+        subprocess.check_call(["make", "containers"], cwd=projroot)
+        subprocess.check_call(["make", "run"], cwd=projroot)
+
+    def tearDown(self):
+        # Make this only on ubuntu-22.04
+        version = subprocess.check_output(["podman", "-v"]).decode().split(' ')[-1]
+        if version[0] == '4':
+            subprocess.check_call(["make", "clean"], cwd=projroot)
+        else:
+            subprocess.check_call(["make", "clean-test"], cwd=projroot)
```

Comment:
Please check version < 4, so that on 5 and newer it also uses the better thing. But moreover, this does not belong here, but in the Makefile itself. `make clean` should always work.

Reply:
Since podman < 4 seems basically doomed anyway, let's not bother.